proxy/auth.go

```diff
@@ -11,16 +11,53 @@ import (
 
 //SetAuth sets basic auth for the given gateway
 func SetAuth(req *http.Request, gateway string) {
-	username, password, err := config.LookupAuthConfig(gateway)
+	authConfig, err := config.LookupAuthConfig(gateway)
+	if err != nil {
+		// no auth info found
+		return
+	}
+	username, password, err := config.DecodeAuth(authConfig.Token)
 	if err != nil {
 		// no auth info found
 		return
 	}
-
 	req.SetBasicAuth(username, password)
 }
 
 //SetToken sets authentication token
 func SetToken(req *http.Request, token string) {
 	req.Header.Set("Authorization", "Bearer "+token)
 }
+
+//SetBasicAuth set basic authentication
+func SetBasicAuth(req *http.Request, authConfig config.AuthConfig) {
+	username, password, err := config.DecodeAuth(authConfig.Token)
+	if err != nil {
+		// no auth info found
+		return
+	}
+	req.SetBasicAuth(username, password)
+}
+
+//SetOauth2 set oauth2 token
+func SetOauth2(req *http.Request, authConfig config.AuthConfig) {
+	SetToken(req, authConfig.Token)
+}
+
+//AddAuth add authentication
+func AddAuth(req *http.Request, gateway string) {
+	authConfig, err := config.LookupAuthConfig(gateway)
+	if err != nil {
+		// no auth info found
+		return
+	}
+
+	switch authConfig.Auth {
+	case config.BasicAuthType:
+		SetBasicAuth(req, authConfig)
```

Comment:
with this implementation, it seems the the `SetBasicAuth` and `SetOauth2` methods could be private.  If we are willing to change the signature, we could pass  the token

```
func AddAuth(req *http.Request, gateway string, token string) {}
```

and set the token auth here too, if it is non-empty.  But that would be reasonable to do in a future PR.